utils/build/docker/nodejs/nextjs/src/app/tag_value/[tag]/[status]/route.js

```diff
@@ -8,6 +8,42 @@ export async function GET (request, { params }) {
 
   rootSpan?.setTag('appsec.events.system_tests_appsec_event.value', params.tag)
 
-  return NextResponse.json('Value tagged')
+  // read querystring
+  const query = Object.fromEntries(new URL(request.url).searchParams)
+
+  // always read body for API sec
+  const body = await getBody(request)
+
+  let response
+
+  if (params?.tag?.startsWith?.('payload_in_response_body') && request.method === 'POST') {
+    response = { payload: body }
+  } else {
+    response = 'Value tagged'
+  }
+
+  return NextResponse.json(response, {
+    status: params.status || 200,
+    headers: new Headers(query)
+  })
 }
 export const OPTIONS = GET
+export const POST = GET
+
+// this function works but only when formData() is called before json()
+// because formData doesn't try to read the body if it's not has form data headers
+async function getBody (request) {
+  try {
+    return Object.fromEntries(await request.formData())
+  } catch (err) {}
+
+  try {
+    return await request.json()
+  } catch (err) {}
+
+  try {
+    return await request.text()
+  } catch (err) {}
```

Comment:
So then this will never work if json() consumes the body no ?

Reply:
I removed it, and the test works in local, let's see if it works in the ci.

Reply:
lgtm